components/Chat/Chat.vue

```diff
@@ -751,9 +751,21 @@
         voice.rate   = this.getTtsRate / 10.0;
         voice.volume = this.getTtsVolume / 10.0;
         voice.pitch  = pitch;
-        // Completely sanitized messages (i.e. emotes only) don't get read
-        // TODO: implement empty message filter
-        voice.text   = ( this.getTtsReadUsername ? `${username} says: ` : '' ) + message;
+        voice.text   = message;
+
+        if( this.getTtsReadUsername ) {
+
+          // Since 'username says' will be prepended to the read message, we have to ensure
+          // that the message isn't blank; skip only if there isn't readable text.
+          const pattern = new RegExp( '\S', 'gi' );
+          if( !message.match( pattern ) ) {
+            console.log( 'Empty message, skipping TTS.' );
+            return;
+          }
+
+          voice.text = `${username} says: ` + voice.text;
```

Comment:
I'm not a fan of template literals mixed with string concatenation, though I can see why you formatted this in the way you did here.

I would just do: `voice.text = \`${username} says: ${message}\`;`but I'll leave the formatting up to you on this one.